pipelines/controller/src/main/java/com/google/fhir/analytics/PipelineManager.java

```diff
@@ -562,21 +573,23 @@ public void run() {
       }
     }
 
+    // TODO fix the potential race condition here; for example if the pipeline finishes
+    //   at the same time that a `POST /tables` request is being handled by the controller.
```

Comment:
Yeah it is unlikely but our current checks do not make it impossible. Here is an [unlikely] scenario: User sends `POST /tables` request and while the tables are being created a pipeline run request comes in as well. If the run happens quickly, it will then start making tables (and updating canonical ones) again. I think there is nothing to make the two mutually exclusive at that point.

Reply:
BTW, this issue is mitigated in PR #811 by sharing the `HiveTableManager` instance and making its `createResourceAndCanonicalTables` synchronized. So I can probably remove this TODO in the next PR.